### DISCUSSION TITLE
Is there no way to create an assistant and share it across a team?

### DISCUSSION THREAD

**Discussion by meltedhead** [infiniflow/ragflow#3364] [Q&A] [answered]
I have been testing with the demo version and i have been able to share the knowledge base with another user but I can't seem to share the assistant i have created with another user? I can't find anyway to do this via the GUI? Is there a way to do that?

**Comment by JinHai-CN** [comment, marked as answer]
No, assistant can't be shared, right now.

**Reply by meltedhead** [reply]
Thanks, so i would need to create the same assistant for each user via the API? Is this something that's planned? Ideally, we would be interested in locking this down so that admin controls this, as users may not have enough knowledge to create assistants that are correct for the data they interact with.

**Reply by JinHai-CN** [reply]
The LLM associated with the assistant will also be available to other users. This can be sensitive, so we're thinking about how to implement it better.  What do you suggest?

**Reply by meltedhead** [reply]
For our use case, we'll be using private LLMs exclusively within our user base, so we don't need to worry too much about data sensitivity between users. That said, I can understand how this could be a concern for others in environments that aren’t private. Ideally, we would set up a private LLM for the platform and make it available to all users, with access to other LLMs and models restricted—only admins would have the ability to create connections to external models. From there, we could introduce the idea of a public chat assistant/agent, which would be available to all users based on the controlled model setup. Additionally, users could opt for a private assistant if they want to create their own.

**Reply by meltedhead** [reply]
Public chat assistants could be restricted to creation by admins only and would only be available in environments where the LLM is set up locally or privately.

**Reply by meltedhead** [reply]
It would be great if the same also applied for Agent workflows, can these be pushed for all users? or do we need to create a template ourselves like what you have and push that to all users?

**Reply by meltedhead** [reply]
Ideally, I'd like to develop a component and integrate it into your code that enables users to run code on a SAS server. From reviewing the existing code, it seems like this would be fairly straightforward. We could then set up a template, similar to your text-to-SQL agent template, for this functionality. If we create this, would it be possible to push it to all users via the API?
